lib/ex_aws/rekognition.ex

```diff
@@ -2,37 +2,126 @@ defmodule ExAws.Rekognition do
   @moduledoc """
   Operations on ExAws Rekognition
   """
+  alias S3Object
 
   @actions %{
+    compare_faces: :post,
+    delete_faces: :post,
+    detect_faces: :post,
+    create_collection: :post,
+    delete_collection: :post,
+    describe_collection: :post,
+    list_collections: :post,
     detect_text: :post
   }
 
-  def detect_text(image) when is_binary(image) do
-    image_64 =
-      image
-      |> Base.encode64()
+  @doc """
+  https://docs.aws.amazon.com/rekognition/latest/dg/API_CompareFaces.html
+  """
+  @spec compare_faces(number(), binary() | S3Object.t(), binary() | S3Object.t()) :: %{
+          optional(any) => any
+        }
+  def compare_faces(similarity_threshold \\ 0.8, source_image, target_image)
+      when is_number(similarity_threshold) do
+    request(:compare_faces, %{
+      "SimilarityThreshold" => similarity_threshold,
+      "SourceImage" => map_image(source_image),
+      "TargetImage" => map_image(target_image)
+    })
+  end
 
-    data = %{
-      "Image" => %{
-        "Bytes" => image_64
-      }
-    }
+  @doc """
+  https://docs.aws.amazon.com/rekognition/latest/dg/API_DeleteFaces.html
+  """
+  @spec delete_faces(binary(), maybe_improper_list()) :: %{optional(any) => any}
+  def delete_faces(collection_id, face_ids) when is_binary(collection_id) and is_list(face_ids) do
+    request(:delete_collection, %{
+      "CollectionId" => collection_id,
+      "FaceIds" => face_ids
+    })
+  end
+
+  @doc """
+  https://docs.aws.amazon.com/rekognition/latest/dg/API_DetectFaces.html
 
-    request(:detect_text, data)
+  NOTE: When using an S3Object, you may need to insure that
+  the S3 uses the same region as Rekognition
+  """
+  @spec detect_faces(maybe_improper_list(), binary() | S3Object.t()) :: %{optional(any) => any}
+  def detect_faces(attributes \\ ["DEFAULT"], image) when is_list(attributes) do
```

Comment:
Save as above, `attributes` have a default value, and should be the last argument. `image` is more usefull as first argument, to be used with `|>`.